src/resources/lib/modules/connman.py

```diff
@@ -1042,15 +1046,43 @@ def dbus_error_handler(self, error):
         try:
             self.oe.dbg_log('connman::dbus_error_handler', 'enter_function', 0)
             self.oe.set_busy(0)
+            if self.reconn_attempt > 1:
+                self.reconn_attempt = 0
             err_name = error.get_dbus_name()
             if 'InProgress' in err_name:
-                self.disconnect_network()
+                if self.user_disconn != 1:
+                    self.disconnect_network()
+                else:
+                    self.user_disconn = 0
                 self.connect_network()
             else:
                 err_message = error.get_dbus_message()
-                if not 'Did not receive a reply' in err_message:
+                if 'Operation aborted' in err_message or 'Input/output error' in err_message:
+                    if self.reconn_attempt < 2:
+                        self.log_error = 0
+                        self.notify_error = 0
+                        self.reconn_attempt += 1
+                        time.sleep(5)
+                        self.connect_network()
+                    else:
+                        self.log_error = 1
+                        self.notify_error = 1
+                elif 'Did not receive a reply' in err_message:
+                    self.log_error = 1
+                    self.notify_error = 0
+                else:
+                    self.log_error = 1
+                    self.notify_error = 1
+                    self.reconn_attempt = 0
+                if self.notify_error == 1:
                     self.oe.notify('Network Error', err_message)
-                self.oe.dbg_log('connman::dbus_error_handler', 'ERROR: (' + err_message + ')', 4)
+                else:
+                    self.notify_error = 1
+                if self.log_error == 1:
+                    self.oe.dbg_log('connman::dbus_error_handler', 'ERROR: (' + err_message + ')', 4)
+                else:
+                    self.log_error = 1
+
```

Comment:
Before this commit is merged i would like to remove this empty line i added by mistake, i will leave it alone until it has bin tested.